roles/server/tasks/main.yml

```diff
@@ -69,7 +70,9 @@
     mode: "{{ item.directory_mode | default('0775') }}"
     setype: "{{ item.setype | default('samba_share_t') }}"
   loop: "{{ samba_shares }}"
-  when: samba_shares is defined and samba_shares | length > 0
+  when: ( samba_shares is defined and samba_shares | length > 0 ) and
+        ( samba_manage_directories is defined and samba_manage_directories ) and
+        ( item.manage_directory is undefined or item.manage_directory != false )
```

Comment:
I don't think it's a good idea of having both root-level and share-level options to manage directories. It complicates things unnecessarily. It either creates the directories or not. So I would remove this line.

In addition, can you please add documentation to the README.md (under Server Role Variables)?